lib/vdsm/tool/configurators/multipath.py

```diff
@@ -327,44 +327,60 @@ def configure():
     commands.run([_MULTIPATHD.cmd, "reconfigure"])
 
 
-def isconfigured():
+def _check_conf_rev(mpathconf):
     """
-    Check the multipath daemon configuration. The configuration file
-    /etc/multipath.conf should contain a tag in form
+    The multipath configuration file should contain a tag in form
     "RHEV REVISION X.Y" for this check to succeed.
     If the tag above is followed by tag "RHEV PRIVATE" the configuration
     should be preserved at all cost.
+
+    Args:
+        mpathconf (List[str]): Lines in the multipath daemon
+        configuration file.
+
+    Returns:
+        bool: check has passed.
+    """
+    first = mpathconf[0]
+    second = mpathconf[1]
+    if _PRIVATE_TAG in second or _OLD_PRIVATE_TAG in second:
+        sys.stdout.write("Manual override for multipath.conf detected"
+                         " - preserving current configuration\n")
+        if _CURRENT_TAG not in first:
+            sys.stdout.write("This manual override for multipath.conf "
+                             "was based on downrevved template. "
+                             "You are strongly advised to "
+                             "contact your support representatives\n")
+        return True
+
+    if _CURRENT_TAG in first:
+        sys.stdout.write("Current revision of multipath.conf detected,"
+                         " preserving\n")
+        return True
+
+    for tag in _OLD_TAGS:
+        if tag in first:
+            sys.stdout.write("Downrev multipath.conf detected, "
+                             "upgrade required\n")
+            return False
+
+    sys.stdout.write("No revision of multipath.conf detected.\n")
+    return False
+
+
+def isconfigured():
+    """
+    Check the multipath daemon configuration at /etc/multipath.conf.
     """
 
     if os.path.exists(_CONF_FILE):
-        first = second = ''
         with open(_CONF_FILE) as f:
             mpathconf = [x.strip("\n") for x in f.readlines()]
         try:
-            first = mpathconf[0]
-            second = mpathconf[1]
+            conf_check = _check_conf_rev(mpathconf)
+            return YES if conf_check else NO
         except IndexError:
```

Comment:
Do we want to fail on the first check or do all the checks and report all errors?
Reporting all errors at once may be more user friendly but harder to get right.

If we want to check all, and keep the results, lets use more clear names:

    revision_ok = _check_revision(mpathconf)
    user_friendly_names_ok = _check_user_friendly_names(mpathconf)

    report errors...